vcf2fhir/fhir_helper.py

```diff
@@ -325,9 +360,45 @@ def add_variant_obv(self, record, ref_seq, ratio_ad_dp):
             )
         observation_dv.component.append(observation_dv_component2)
 
-        if alleles['CODE'] != "" or alleles['ALLELE'] != "":
+        if genomic_source_class in ['germline', 'somatic']:
+            source_class = get_genomic_source_class(genomic_source_class)
             observation_dv_component3 = observation.ObservationComponent()
             observation_dv_component3.code = concept.CodeableConcept(
+                {
+                    "coding": [
+                        {
+                            "system": "http://loinc.org",
+                            "code": "48002-0",
+                            "display": "Genomic Source Class"
+                        }
+                    ]
+                }
+            )
+            observation_dv_component3\
+                .valueCodeableConcept = concept.CodeableConcept(
+                    {
+                        "coding": [
+                            {
+                                "system": "http://loinc.org",
+                                "code": source_class["CODE"],
+                                "display": source_class["DISPLAY"]
+                            }
+                        ]
+                    }
+                )
+            observation_dv.component.append(observation_dv_component3)
+
+        if(
+                (alleles['CODE'] != "" or alleles['ALLELE'] != "") and
+                genomic_source_class == 'germline' and
+                not (record.is_sv and record.INFO['SVTYPE'] not in [
+                                                                        'INS',
+                                                                        'INV',
+                                                                        'DUP',
+                                                                        'DEL'
+                                                                    ])):

```

Comment:
If conditions in if case become such huge it's a good practice to add a comment before it. It helps the reader to understand what `if` is trying to check.